src/abstractdataframe/subset.jl

```diff
@@ -0,0 +1,231 @@
+# subset allows a transformation specification without a target column name or a column
+
+_process_subset_pair(i::Int, a::ColumnIndex) = a => Symbol(:x, i)
+_process_subset_pair(i::Int, @nospecialize(a::Pair{<:Any, <:Base.Callable})) =
+    first(a) => last(a) => Symbol(:x, i)
+_process_subset_pair(i::Int, a) =
+    throw(ArgumentError("condition specifier $a is not supported by `subset`"))
+
+function _get_subset_conditions(df::Union{AbstractDataFrame, GroupedDataFrame},
+                               @nospecialize(args), skipmissing::Bool)
+    conditions = Any[_process_subset_pair(i, a) for (i, a) in enumerate(args)]
+
+    isempty(conditions) && throw(ArgumentError("at least one condition must be passed"))
+
+    if df isa AbstractDataFrame
+        df_conditions = select(df, conditions..., copycols=!(df isa DataFrame))
+    else
+        df_conditions = select(df, conditions...,
+                               copycols=!(parent(df) isa DataFrame), keepkeys=false)
+    end
+    test_type = skipmissing ? Union{Missing, Bool} : Bool
+    for col in eachcol(df_conditions)
+        if !(eltype(col) <: test_type)
+            throw(ArgumentError("each transformation must produce a vector whose " *
+                                "eltype is subtype of $test_type"))
+        end
+    end
+    @assert ncol(df_conditions) == length(conditions)
+    if ncol(df_conditions) == 1
+        return .===(df_conditions[!, 1], true)
+    else
+        return .===(.&(eachcol(df_conditions)...), true)
+    end
+end
+
+"""
+    subset(df::AbstractDataFrame, args...; skipmissing::Bool=false, view::Bool=false)
+    subset(gdf::GroupedDataFrame, args...; skipmissing::Bool=false, view::Bool=false)
+
+Return a copy of data frame `df` or parent of `gdf` containing only rows for
+which all values produced by transformation(s) `args` for a given row is `true`.
+
+If `skipmissing=true` returing `missing` in `args` is allowed and these rows are
```

Comment:
```suggestion
If `skipmissing=true`, returning `missing` in `args` is allowed and corresponding rows are
```
Also, maybe better mention the default behavior first?